include/base/libmesh_common.h

```diff
@@ -634,6 +634,8 @@ inline Tnew libmesh_cast_int (Told oldvar)
 template <class T>
 constexpr std::false_type always_false{};
 
+static constexpr std::size_t max_mesh_dim = LIBMESH_DIM;
```

Comment:
I really am not tied to this. `libmesh_dim` sounds great

Reply:
`libMesh::libmesh_dim` looks a little funny but as most code using this will be in the libmesh namespace I suppose that's fine. Even if we use it in MOOSE we just have everything globally included from the libMesh namespace (to my own sadness) everywhere so we wouldn't see that

Reply:
How about `libMesh::max_dim`?  I guess when we've used `libmesh_` in the past it's always either been a macro or a C function (where we can't benefit from the namespace) or something like `libmesh_real` where we've really wanted to make it clear that this is a custom shim to something that's already in `std::`, just made to handle other types or other C++ versions too.

Reply:
I started with `max_dim` and I think it conflicted with something else, at least I believe that's the reason for https://github.com/libMesh/libmesh/pull/3883/commits/1281eb248674332e1a7c4670244c18439a172da5

Reply:
Ah, damn.  Yeah, we use `max_dim` locally in three different files, and I'll bet you got shadowing warnings from all three.  Well, my vote's for `libMesh::libmesh_dim` (which ... sort of makes sense? we have lots of `dim` dimensions in libMesh, but the one that limits all our code is the *libmesh* dim?), but I'm happy with `max_mesh_dim` (which *is* that, even if it's also a max elem dim and max `TypeVector` dim and so on) too.  We could bug John for a tie breaker or you could just do whatever you want.  :-D